### PR TITLE
Add naming convention config dummies

### LINKED ISSUE #1454: Make config of solver dummies follow conventions
https://github.com/precice/precice/blob/develop/examples/solverdummies/precice-config.xml

does not yet follow the conventions we use for tutorials:

https://precice.org/community-contribute-to-precice.html#naming-conventions

Would be nice to have these consistent.

### PR DESCRIPTION
**Pull request by MarkAshraf96**
## Main changes of this PR

- Modified  the solver dummies configuration file to meet the naming convention.

closes #1454 

## Author's checklist

* [ ] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I added a test to cover the proposed changes in our test suite.
* [ ] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).
* [ ] I sticked to C++17 features.
* [ ] I sticked to CMake version 3.16.3.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

**Comment by IshaanDesai** [comment]
Please have a look at the failing tests and make the necessary changes so that they pass.

**Comment by MarkAshraf96** [comment]
@IshaanDesai Hello Ishaan, Can you approve running the workflows for me. Thank you ind advance.

File: examples/solverdummies/precice-config.xml [hunk 1]
```diff
@@ -10,40 +10,40 @@
...
+    <mesh name="Mesh-Two">
+      <use-data name="Data-One" />
+      <use-data name="Data-Two" />
     </mesh>
 
     <participant name="SolverOne">
-      <provide-mesh name="MeshOne" />
-      <write-data name="dataOne" mesh="MeshOne" />
-      <read-data name="dataTwo" mesh="MeshOne" />
+      <provide-mesh name="SolverOne-SolverTwo-Mesh-One" />
```
[the remark this answers is not in the document]
Comment (MarkAshraf96) [reply examples/solverdummies/precice-config.xml]: Thank You for your suggestion I applied it, I thought the the 5th point in the naming conventions means that the solver (interface) should be included in the mesh name.

**Comment by MarkAshraf96** [comment]
I believe that the tests are hard-coded to the variables name that already exists in the file, and maybe the tests also need to be changed to match with the naming convention.

**Comment by MarkAshraf96** [comment]
> I believe that the tests are hard-coded to the variables name that already exists in the file, and maybe the tests also need to be changed to match with the naming convention.

@uekerman

**Comment by IshaanDesai** [comment]
> > I believe that the tests are hard-coded to the variables name that already exists in the file, and maybe the tests also need to be changed to match with the naming convention.
> 
> @uekerman

Changing to tests to whatever is being changed in the pull request is a standard task. if something is hard-coded in the tests then naturally it also has to change.

**Comment by MarkAshraf96** [comment]
@uekerman A kind reminder that my request is waiting for the workflows to run.

**Comment by IshaanDesai** [comment]
> A kind reminder that my request is waiting for the workflows to run.

As per what you wrote in https://precice.discourse.group/t/installing-precice-on-mac-problem-with-eigen-library/1235/9, preCICE is compiling on your machine, right? You can now run the tests locally and see if everything works: https://precice.org/installation-source-testing.html

**Review by fsimonis — COMMENTED** [review]
Given that these changes are user-facing, we need a changelog entry for these changes.

@IshaanDesai Are you aware of the state of the solver dummies in the binding repositories?

**Comment by IshaanDesai** [comment]
> Are you aware of the state of the solver dummies in the binding repositories?

Will open an issue in the Python bindings repo.

File: examples/solverdummies/precice-config.xml [hunk 2]
```diff
@@ -10,40 +10,40 @@
...
   <solver-interface dimensions="3">
-    <data:vector name="dataOne" />
-    <data:vector name="dataTwo" />
+    <data:vector name="Data-One" />
+    <data:vector name="Data-Two" />
 
-    <mesh name="MeshOne">
-      <use-data name="dataOne" />
-      <use-data name="dataTwo" />
+    <mesh name="Mesh-One">
```
Comment (uekerman) [inline examples/solverdummies/precice-config.xml:16]:
```suggestion
    <mesh name="SolverOne-Mesh">
```

as stated in the [conventions](https://precice.org/community-contribute-to-precice.html#naming-conventions):

> Mesh names start with the participant/domain name, e.g. Fluid-Mesh

**Review by uekerman — CHANGES_REQUESTED** [review]

File: examples/solverdummies/precice-config.xml (hunk 2 above)
Comment (MarkAshraf96) [reply examples/solverdummies/precice-config.xml]: re modified the mesh names to include the solver name in #[2301cb4](https://github.com/precice/precice/pull/1548/commits/2301cb4c9ce6ade0b84562ba92b3b6c432c61ac6)

File: docs/changelog/1548.md [hunk 3]
```diff
@@ -0,0 +1 @@
+- Modified Data, and Mesh names to match the naming convention.
```
Comment (uekerman) [inline docs/changelog/1548.md:1]:
```suggestion
- Modified data and mesh names in solverdummy configuration to match naming conventions.
```

**Review by uekerman — APPROVED** [review]
`Data-One` vs `SolverOne` is still a bit odd. @MakisH opinions?
Otherwise good to go :+1:

Let's not forget to also adjust all downstream bindings.

**Pull request merged**